### PR TITLE
enforce authentication realm for dockerhub login

### PR DESCRIPTION
**Pull request by fahedouch**
point dockerhub login addresses to the [authentication legacy default server ](https://github.com/containerd/nerdctl/blob/main/pkg/imgutil/dockerconfigresolver/dockerconfigresolver_util.go#L33)

fixing https://github.com/containerd/nerdctl/issues/3245

### CODE REVIEW

**Comment by apostasie** [comment]
This is a good short term fix.
I think we should merge this regardless.

However, it will save independent credentials for different variants, which it should not (because normalization should happen before):
```
{
	"auths": {
		"docker.io": {
		},
		"https://index.docker.io/v1/": {
		},
		"index.docker.io": {
		}
	}
}
```

**Comment by fahedouch** [comment]
> This is a good short term fix. I think we should merge this regardless.
> 
> However, it will save independent credentials for different variants, which it should not (because normalization should happen before):
> 
> ```
> {
> 	"auths": {
> 		"docker.io": {
> 		},
> 		"https://index.docker.io/v1/": {
> 		},
> 		"index.docker.io": {
> 		},
> 		"index.docker.io:443": {
> 		},
> 		"registry-1.docker.io": {
> 		}
> 	}
> }
> ```

good catch. 

Now authentification endpoints are more straightforward; they are redirected to the authentication realm.

File: pkg/cmd/login/login.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func GetDefaultAuthConfig(checkCredStore bool, serverAddress string, isDefaultRe
 			return nil, err
 		}
 	}
-	var authconfig = dockercliconfigtypes.AuthConfig{}
+	authconfig := dockercliconfigtypes.AuthConfig{}
```
Comment (AkihiroSuda) [inline pkg/cmd/login/login.go:127]: Probably these cosmetic changes should be a separate commit

Comment (fahedouch) [reply pkg/cmd/login/login.go:127]: is it the gofumpt

**Review by AkihiroSuda — APPROVED** [review]
Thanks
